### PR TITLE
[ENH] Render column headers and values as raw string

### LINKED ISSUE #298: ensure that the column headers and values are rendered as raw strings with no escaping (so that the values are correctly visible)
As a user, I want to see the raw string of column names and cell values, so that I can easily spot things like empty strings (`""`), spaces (`"myvalue "`), or strange whitespace characters (`"hello\r"`).

TODO:

- [x] Ensure all column headers and cell values are clearly shown as raw strings
- [x] Ensure empty values and whitespaces are clearly visible (including newline etc)
- [x] Write a test to ensure that trailing whitespace etc is not stripped (component/unit) and correctly displayed (e2e)

Cases I would like handled:

- [x] empty: `""`
- [x] space: `" "`
- [x] trailing space: `"word "`
- [x] Line endings: `"end\n"`, `"end\r"`, `"end\r\n"`

### PR DESCRIPTION
**Pull request by rmanaem**
- Closes #298

Changes proposed in this pull request:

- Applied JSON.stringify to column names and cell values in UI components to ensure whitespace (spaces) and newlines (\n, \r) are preserved and visible as raw strings
- Replaced logical OR (||) with nullish coalescing (??) when resolving column names. This prevents valid empty string names from being incorrectly evaluated as falsy and falling back to internal column IDs
- Added e2e and component test coverage

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [ ] Checks pass
- [ ] If the PR changes the default [Neurobagel `config.json` or any of the term files](https://github.com/neurobagel/communities/tree/main/configs/Neurobagel), make sure their respective counterparts in [neurobagel/communities](https://github.com/neurobagel/communities) have been updated accordingly.

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

## Summary by Sourcery

Render table column headers, names, and values as raw strings so whitespace and newline characters are visibly preserved across the annotation workflow.

Enhancements:
- Preserve column display names when they are empty strings by using nullish coalescing instead of logical OR when resolving column names.
- Render column headers, column labels, and cell values using JSON-style stringification so spaces and newline characters are shown explicitly throughout the UI.

Tests:
- Add component tests for ValueTable to verify whitespace and newline characters are displayed as raw strings.
- Add an end-to-end test and TSV fixture to validate whitespace rendering from data upload through column and value annotation views.

### CODE REVIEW

File: src/components/ColumnTypeCollapse.tsx [hunk 1]
```diff
@@ -288,7 +289,7 @@ function ColumnTypeCollapse({
                             }}
                           >
                             <Typography sx={{ flexGrow: 1, fontSize: '0.9rem' }}>
-                              {entry.column.name || entry.id}
+                              <VisibleWhitespace value={entry.column.name ?? entry.id} />
```
Comment (surchs) [inline src/components/ColumnTypeCollapse.tsx]: am I missing something or could you not just reuse dictionaryKey here?

File: src/components/VisibleWhitespace.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      <span data-cy="visible-whitespace" className="whitespace-pre-wrap font-mono">
+        &quot;&quot;
+      </span>
+    );
+  }
+
+  const visuallyFormatted = value
+    .replace(/ /g, '·')
+    .replace(/\t/g, '→')
+    .replace(/\n/g, '¶\n')
```
Comment (surchs) [inline src/components/VisibleWhitespace.tsx]: why both `¶` and `\n`? seems like `¶` would be sufficient?

File: cypress/component/VisibleWhitespace.cy.tsx
Comment (surchs) [inline cypress/component/VisibleWhitespace.cy.tsx:1]: good tests - I wonder though if this wouldn't make more sense as a unit test for a utility function that just does the string re-formatting. wdyt?

**Review by surchs — COMMENTED** [review]
Thanks @rmanaem for working on this! The behaviour looks good to me, but before I approve, could you remind me what the use case is for having the "highlight whitespace" toggle turned off? The context provider seems easy enough for it, but there is a good bit of test coverage to ensure the toggle works - and there is now also the toggle in the UI that a user must understand. I know we discussed this, but it seems to me just rendering whitespace with dots and newlines could always be the right solution.

File: src/components/VisibleWhitespace.tsx (hunk 2 above)
Comment (rmanaem) [reply src/components/VisibleWhitespace.tsx]: Keeping the `\n` after `¶` so we get the visual line break in the UI

File: cypress/component/VisibleWhitespace.cy.tsx
Comment (rmanaem) [reply cypress/component/VisibleWhitespace.cy.tsx:1]: Good call

File: cypress/fixtures/examples/whitespace_table.tsv [hunk 3]
```diff
@@ -0,0 +1,7 @@
+participant_id		 	diagnosis 	"session
```
Comment (surchs) [inline cypress/fixtures/examples/whitespace_table.tsv:1]: what's going on with this file? It seems like there is an odd mix of spaces, tabs, and quotes. Is that intentional? I.e. it seems like there is no `"` character that is actually closed in this table. Is this meant to be kind of a worst-case table where most everything is wrong?

File: src/components/DataTablePreview.tsx [hunk 4]
```diff
@@ -53,7 +53,7 @@ function DataTablePreview({ dataTable }: { dataTable: DataTable }) {
                     const cellId = uuidv4();
                     return (
                       <TableCell key={cellId} align="left">
-                        {cell}
+                        {cell as React.ReactNode}
```
Comment (surchs) [inline src/components/DataTablePreview.tsx]: could you instead type `rowData` as an array of arrays of strings? i.e. `string[][]`? This type here silences TS but is also a bit broad and uninformative

File: src/components/ValueAnnotation.tsx [hunk 5]
```diff
@@ -21,6 +32,7 @@ function ValueAnnotation() {
   const [selectedColumnIds, setSelectedColumnIds] = useState<string[]>([]);
   const [activeColumnId, setActiveColumnId] = useState<string | null>(null);
   const navData = useValueAnnotationNavData();
+  const [showFormattingMarks, setShowFormattingMarks] = useState(false);
```
Comment (surchs) [inline src/components/ValueAnnotation.tsx]: why does this component own the "show form" state? Doesn't that better belong to the provider / provider-hook?

**Review by surchs — APPROVED** [review]
Thanks @rmanaem for doing these rounds of changes.
Behaviour looks great, I left some minor comments for you to look over. But ready to merge

🧑‍🍳

File: cypress/fixtures/examples/whitespace_table.tsv (hunk 3 above)
Comment (rmanaem) [reply cypress/fixtures/examples/whitespace_table.tsv:1]: Yes, this file is supposed to have a case of every possible whitespace we might run into for the e2e test that we wrote for the new feature.

File: src/components/DataTablePreview.tsx (hunk 4 above)
Comment (rmanaem) [reply src/components/DataTablePreview.tsx]: Good catch and good call, will do.
